soc/arm/nordic_nrf/nrf52/power.c

```diff
@@ -115,3 +115,11 @@ bool _sys_soc_is_valid_power_state(enum power_states state)
 
 	return false;
 }
+
+/* Overrides the weak ARM implementation:
+   Set general purpose retention register and reboot */
+void sys_arch_reboot(int type)
+{
+	nrf_power_gpregret_set(type);
```

Comment:
I believe nrf_power_gpregret_set() takes an unsigned u8 as argument. So, please, check if passing -int- gives warning/error. Ideally, we should have type-casting here, right?

Reply:
Same comment applies for nRF51